server/src/main/java/com/voxelwind/server/game/level/manager/LevelChunkManager.java

```diff
@@ -76,11 +139,17 @@ void execute() {
 
             // Load the chunk.
             backingChunkProvider.createChunk(level, x, z).whenComplete((chunk, throwable) -> {
+                long chunkKey = toLong(x, z);
                 if (chunk != null) {
-                    chunksLoaded.put(new Vector2i(x, z), chunk);
+                    chunksLoaded.put(chunkKey, chunk);
                 }
+
                 done.set(true);
-                chunksToLoad.remove(new Vector2i(x, z));
+                chunksToLoad.remove(chunkKey);
+                long current = System.currentTimeMillis();
+                loadedTimes.put(chunkKey, current);
+                lastAccessTimes.put(chunkKey, current);
```

Comment:
Load and access times should _only_ be updated if the chunk was successfully loaded.